ast/edit/task_editor.go

```diff
@@ -80,6 +102,21 @@ func GetProperty(obj *ast.ObjectExpression, key string) (ast.Expression, error)
 	}
 }
 
+// GetAllProperties returns all properties for a given option
+func GetAllProperties(obj *ast.ObjectExpression) ([]ast.Expression, error) {
```

Comment:
I'm looking at the `ast.Expression` type, which looks like an extension interface around the `ast.Node` type, which looks like:

```go
// Node represents a node in the InfluxDB abstract syntax tree.
type Node interface {
	node()
	Type() string // Type property is a string that contains the variant type of the node
	Location() SourceLocation
	Errs() []Error
	Copy() Node

	// All node must support json marshalling
	json.Marshaler
}
```

how does one pull out the key of the option properties from the returned `[]ast.Expression`?

Reply:
`ast.Expression` types can be type converted to their underlying ast Node type. [Something like this](https://github.com/influxdata/flux/blob/master/ast/edit/task_editor_test.go#L24) on lines 24 and 29. If the type conversion fails, then you're converting it to the wrong ast Node type.

Reply:
[Here's a place where I'm assuming types within the api.](https://github.com/influxdata/flux/blob/master/ast/edit/task_editor.go#L9-L28)

Reply:
ah ok. I think I'm following. Do I look at the `Type` method on Node do determine which type assert to use, or a type switch?

Reply:
yeah, I _think_ you could call the `Type()` method if you don't already know the Node type and then assert. You could switch on the result of the called to `Type()` if you have a predetermined list of possible types.